lib/CL/pocl_llvm_api.cc

```diff
@@ -534,54 +534,112 @@ static int pocl_llvm_get_kernel_arg_metadata(const char* kernel_name,
                                              llvm::Module *input,
                                              cl_kernel kernel)
 {
-
-  // find the right kernel in "opencl.kernels" metadata
-  llvm::NamedMDNode *opencl_kernels = input->getNamedMetadata("opencl.kernels");
   llvm::MDNode *kernel_metadata = NULL;
+  int bitcode_is_spir = input->getTargetTriple().find("spir") == 0;
+  SmallVector<std::pair<std::string, llvm::MDNode*>, 6> llvm39_kernel_meta;
+  bool llvm_older_than_3_9 = false;
```

Comment:
Why a variable, why you cannot simply if (LLVM_OLDER_THAN_3_9) and define to 1/0?